src/main/java/de/tum/cit/aet/artemis/programming/repository/VcsAccessLogRepository.java

```diff
@@ -27,21 +27,36 @@
 public interface VcsAccessLogRepository extends ArtemisJpaRepository<VcsAccessLog, Long> {
 
     /**
-     * Find the access log entry which does not have any commit hash yet
+     * Retrieves the most recent {@link VcsAccessLog} for a given participation ID.
      *
-     * @param participationId The id of the participation the repository belongs to
-     * @return a log entry belonging to the participationId, which has no commit hash
+     * @param participationId the ID of the participation to filter by.
+     * @return an {@link Optional} containing the newest {@link VcsAccessLog}, or empty if none exists.
      */
     @Query("""
-
-            SELECT vcsAccessLog
+                SELECT vcsAccessLog
                 FROM VcsAccessLog vcsAccessLog
                 WHERE vcsAccessLog.participation.id = :participationId
-                ORDER BY vcsAccessLog.timestamp DESC
+                ORDER BY vcsAccessLog.id DESC
```

Comment:
Is there a specific reason for that? Should work fine for all our auto-incremented IDs but still curious

Reply:
Not really, I just wanted to use the same ordering method across all retrieval methods. Possibly the sort by id might be a bit faster, but probably it does not make a difference at all